File: random-problems/graphtoarray_python/graph_to_array.py
def graph_to_array(list_node, memo): # right
	if(list_node["next"] == None):
		return [list_node["val"]]
	return [list_node["val"]] + graph_to_array(list_node["next"], memo)

# def graph_to_array(list_node, memo): # wrong
#     if(list_node["next"] == None):
#     	return [list_node["val"]]
#     return memo + graph_to_array(list_node["next"], memo)

File: random-problems/graphtoarray/graphtoarray.js
var graphToArray = function (listNode, memo) {
	//run once
	if(listNode.next == null){
		memo.push(listNode.val)
		return memo;
	}
	//go down
	//but memo here call stacks above the current one
	return [listNode.val].concat(graphToArray(listNode.next, memo)).sort()
};

exports.graphToArray = graphToArray